homeassistant/components/watchdog_file_watcher.py

```diff
@@ -0,0 +1,75 @@
+"""
+Component for monitoring activity on a folder.
+"""
+import asyncio
+import logging
+import voluptuous as vol
+from watchdog.observers import Observer
+from watchdog.events import PatternMatchingEventHandler
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+#DEPENDENCIES = ['watchdog']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PATH = 'folder'
+DOMAIN = "watchdog_file_watcher"
+EVENT_TYPE = "event_type"
+SRC_PATH = "src_path"
+PATTERNS = ["*.txt", "*.py", "*.md", "*.jpg", "*.png"]
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PATH): cv.isdir})
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the folder watcher."""
+    conf = config.get(DOMAIN)
+    path = conf.get(CONF_PATH)
+    if not hass.config.is_allowed_path(path):
+        _LOGGER.error("folder %s is not valid or allowed", path)
+    else:
+        Watcher(path, hass)
+    return True
+
+
+class Watcher(Entity):
+    """Class for starting Watchdog."""
+    def __init__(self, path, hass):
+        self._observer = Observer()
+        self._observer.schedule(
+            MyHandler(hass),
+            path,
+            recursive=True)
+        self._observer.start()
```

Comment:
I think we should only start observing once we're `added_to_hass`. Doing work or I/O in the constructor is a really bad idea 😉

Reply:
There is no `added_to_hass` as it's not an entity. Instead, we should listen to the HOMEASSISTANT_START event and start then.

Reply:
ok